pressbooks/pressbooks-multi-institution
# fix: deactivation

Issue: https://github.com/pressbooks/pressbooks-multi-institution/issues/92

This PR:
- Adds a new command line `wp reset-db-schema` that revokes super admin privileges to all existing institutional managers, then rolls back all the DB migrations and recreates/migrates all the DB schema. (**Attention: all the generated plugin data will be permanently removed if you run this command.**).
- Preserves the DB tables and data upon plugin deactivation.
- Revokes super admin privileges when the plugin is deactivated.
- Looks for all Institutional Managers stored in the DB and grants restricted super admin privileges upon plugin activation.
- Revokes super admin privileges to all the institutional managers assigned to an institution upon institution deletion.

### Testing cases
- Make sure you have institutions, and institutional managers with books and users assigned. You should have access as an institutional manager.
- Deactivate the plugin.
- All the institutional managers should not be super admins anymore.
- Activate the plugin.
- All the institutional managers should now have restricted super admin privileges, just like before deactivation.
- Remove an institution with Institutional Managers.
- All the institutional managers of the removed institution should not be super admins anymore.
- Run `wp reset-db-schema` in the plugin's path and ensure all your data has been permanently removed. (**You will lose all your plugin's data in this step!**)
